### PR TITLE
Refactor/reduce build boilerplate

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

After getting #525 working I realized we'd be running the LRGS multiple times in the automated build (as it's part of the current integration test suite but doesn't depend on any of that infrastructure.)

After deciding it would be best to create yet-another-test-target the amount of duplicate XML that would be required bothered me. Thus I've reduced the boilerplate to make it easier to add new compile and test targets.

## Solution

- Created a `compile-java` macrodef to handle compiling java and dealing with when things are up-to-date
- Created a `test` macrodef to handle running tests.

Callers of the task are required to set the src files for compile, classes for test, and the classpath for both.
Additional jvmargs may be provided for the test tests.
Additional filessets to determine if the output is up-to-date may also be provided.

## how you tested the change

Since the tests pass, the build works. (NOTE: at least for those things under test, but that's now varied enough I feel confident things are in a good state.)

## Where the following done:

- [ ] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

File: src/main/java/lrgs/gui/LrgsBuild.java [hunk 1]
```diff
@@ -19,8 +19,8 @@
 */
 public class LrgsBuild
 {
-	public static final String buildDate = "@DATE@";
-	public static final String rcNum = "@RCNUM@";
-	public static final String version = "@VERSION@";
+	public static final String buildDate = "7.0.13";
```
Comment (ktarbet) [inline src/main/java/lrgs/gui/LrgsBuild.java:22]: Is this, setting specific values, intentional?

Comment (MikeNeilson) [reply src/main/java/lrgs/gui/LrgsBuild.java:22]: Yeah. Though certainly not ideal.

**Review by ktarbet — APPROVED** [review]
